### PR TITLE
Remove PY3.8 support; Add PY3.13 support

### PR DESCRIPTION
**Pull request by ktbyers**
Update GH actions to reflect this.

Switch to `types-setuptools` since pypi removed the previously used types-pkg-resources.

### CODE REVIEW

**Comment by ktbyers** [comment]
Closer, but now breaking at a nbval test (on PY3.13).

**Comment by ktbyers** [comment]
Note, for some reason the docs/tutorial/processors.ipynb test was failing on PY3.13 due to Hosts being in a different order. I assumed this in some way pertained to a threading change i.e. that insertion order was changing.

I just added a `sort_keys=True` on the json.dumps() call to ensure that dictionary order of the hosts was the same (on the output).

**Comment by ktbyers** [comment]
@dbarrosop @ogenstad I think this should be good to go, if one of you want to review/approve.

Note, I will plan on doing a release once this is approved.

Note also, I pinned to a test version of PyEZ for Nornir testing purposes. This test version of PyEZ fixes the underlying PY3.13 breaking issue. This is only in Nornir dev dependencies.

This does imply that anyone using Nornir-napalm and PY3.13 will still run into breakage due to PYEZ, but I don't think we should block all other Nornir users while waiting for PyEZ to fix their underlying issue.

File: nornir/core/configuration.py
Comment (ktbyers) [inline nornir/core/configuration.py:19]: This looked like the recommended way to enforce type consistency i.e. that a Parameter that was a 'str' was consistently required to be a 'str').

There were some mypy issues until I did this.

File: nornir/core/configuration.py
Comment (ktbyers) [inline nornir/core/configuration.py:61]: This was a secondary requirement of using `Parameter(Generic[T])`

File: pyproject.toml
Comment (ktbyers) [inline pyproject.toml:3]: New Poetry build format.

File: pyproject.toml
Comment (ktbyers) [inline pyproject.toml:44]: `types-pkg-resources` was completely removed from pypi--all versions. Was fairly annoyed by that.

**Review by dbarrosop — APPROVED** [review]

**Comment by dbarrosop** [comment]
Great job, thanks!

> This does imply that anyone using Nornir-napalm and PY3.13 will still run into breakage due to PYEZ, but I don't think we should block all other Nornir users while waiting for PyEZ to fix their underlying issue.

Agreed, that shouldn't block nornir core.

**Pull request merged**
